mlir/test/Dialect/Transform/test-pass-application.mlir

```diff
@@ -248,3 +299,77 @@ module attributes {transform.with_named_sequence} {
     transform.yield
   }
 }
+
+// -----
+
+/////////////////////////////////////////////////////////////////////
+// Check that the following cases are caugh in the generic format. //
+/////////////////////////////////////////////////////////////////////
+
+// Invalid due to param_operand_index occurences in options dict not being
+// one-to-one with the dynamic options provided as params:
+//   param_operand_index out of bounds w.r.t. the number of options provided via params.
+
+"builtin.module"() ({
+  "transform.named_sequence"() <{function_type = (!transform.any_op) -> (), sym_name = "__transform_main"}> ({
+  ^bb0(%arg0: !transform.any_op):
+    %0 = "transform.structured.match"(%arg0) <{ops = ["func.func"]}> : (!transform.any_op) -> !transform.any_op
+    %1 = "transform.param.constant"() <{value = 10 : i64}> : () -> !transform.any_param
+    // expected-error @below {{dynamic option index 1 is out of bounds for the number of dynamic options: 1}}
```

Comment:
Why is this restriction necessary?

Reply:
It is zero-based indexing, so it is really an out-of-bounds access as the list of param arguments is a singleton list in this case. Will add a comment to the parser to make this more clear (and the usage of the whole `param_operand_index` attribute).

Reply:
Makes sense, I didn't notice that the last argument is an op so there's only one option to choose from.